java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -107,7 +143,171 @@ BooleanConstraint inverse() {
     }
   }
 
+  public ProgramState processNewInstance(ProgramState initialState, NewClassTree syntaxNode) {
+    final Pair<ProgramState, List<SymbolicValue>> pair = initialState.unstackValue(syntaxNode.arguments().size());
+    ProgramState previousState = pair.a;
+    List<SymbolicValue> arguments = pair.b;
+    final SymbolicValue value;
+    final NullConstraint constraint;
+    if (isOpeningResource(syntaxNode)) {
+      final CloseableArgument closeableArgument = getCloseableArgument(syntaxNode, arguments);
+      if (closeableArgument == null) {
+        value = new SymbolicValue(counter++, syntaxNode);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this increment or decrement operator into a dedicated statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS881)